### PR TITLE
TextPosition.py: Raise exception for zero offset

### PR DESCRIPTION
**Pull request by li-boxuan**
Raise an exception when a line or column offset is zero.

Closes https://github.com/coala/coala/issues/5355

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/results/TextPosition.py [hunk 1]
```diff
@@ -21,6 +23,10 @@ def __init__(self, line: (int, None)=None, column: (int, None)=None):
         if line is None and column is not None:
             raise ValueError('A column can only be set if a line is set.')
 
+        if line == 0 or column == 0:
+            logging.warning('line or column offset should not be 0. '
+                            'Please report this to developers.')
```
Comment (jayvdb) [inline coalib/results/TextPosition.py]: To be useful, we need to report to the user which bear it was.  The only way to do that is raise an exception here, and catch it somewhere higher.

Would be good to have separate exception for line 0 and column 0, to make it easier to find the source of the problem.

File: coalib/results/TextPosition.py [hunk 2]
```diff
@@ -2,6 +2,10 @@
     enforce_signature, generate_ordering, generate_repr)
 
 
+class ZeroOffsetError(Exception):
```
Comment (Makman2) [inline coalib/results/TextPosition.py]: Should be subclassed from `ValueError`, esp. as you do checks on input values and raise it inside `__init__`

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5367/commits/31095951afee08b0123267c579f3d6b673206dca

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**